forc-plugins/forc-fmt-v2/src/main.rs

```diff
@@ -0,0 +1,266 @@
+//! A `forc` plugin for running the Sway code formatter.
+
+use anyhow::{bail, Result};
+use clap::Parser;
+use prettydiff::{basic::DiffOp, diff_lines};
+use std::{
+    default::Default,
+    fs,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use taplo::formatter as taplo_fmt;
+use tracing::{error, info};
+
+use forc_util::{find_manifest_dir, init_tracing_subscriber, println_green, println_red};
+use sway_core::BuildConfig;
+use sway_fmt_v2::{config::manifest::Config, Formatter};
+use sway_utils::{constants, get_sway_files};
+
+#[derive(Debug, Parser)]
+#[clap(
+    name = "forc-fmt",
+    about = "Forc plugin for running the Sway code formatter.",
+    version
+)]
+pub struct App {
+    /// Run in 'check' mode.
+    ///
+    /// - Exits with `0` if input is formatted correctly.
+    /// - Exits with `1` and prints a diff if formatting is required.
+    #[clap(short, long)]
+    pub check: bool,
+    /// Path to the project, if not specified, current working directory will be used.
+    #[clap(short, long)]
+    pub path: Option<String>,
+}
+
+fn main() {
+    init_tracing_subscriber();
+    if let Err(err) = run() {
+        error!("Error: {:?}", err);
+        std::process::exit(1);
+    }
+}
+
+fn run() -> Result<()> {
+    let app = App::parse();
+    let dir = match app.path.clone() {
+        Some(path) => PathBuf::from(path),
+        None => std::env::current_dir()?,
+    };
+    let config = match Config::from_dir(&dir) {
+        Ok(config) => config,
+        Err(_) => Config::default(),
+    };
+    format_pkg_at_dir(app, &dir, config)
+}
+
+/// Format the package at the given directory.
+fn format_pkg_at_dir(app: App, dir: &Path, config: Config) -> Result<()> {
+    match find_manifest_dir(dir) {
+        Some(path) => {
+            let manifest_path = path.clone();
+            let manifest_file = manifest_path.join(constants::MANIFEST_FILE_NAME);
+            let files = get_sway_files(path);
+            let formatting_options = &Formatter::from_opts(config);
+            let mut contains_edits = false;
+
+            for file in files {
+                if let Ok(file_content) = fs::read_to_string(&file) {
+                    let file_content: Arc<str> = Arc::from(file_content);
+                    let build_config = BuildConfig::root_from_file_name_and_manifest_path(
+                        file.clone(),
+                        manifest_path.clone(),
+                    );
+                    match Formatter::format(
+                        formatting_options,
+                        file_content.clone(),
+                        Some(&build_config),
+                    ) {
+                        Ok(formatted_content) => {
+                            if app.check {
+                                if *file_content != *formatted_content {
```

Comment:
`formatted_content` should be fine here, no need to dereference `*formatted_content`